src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -467,4 +485,23 @@ protected String getKeyId(IamPrincipalCredentials credentials) {
 
         return metadata;
     }
+
+    protected Optional<AwsIamRoleRecord> findIamRoleAssociatedWithSdb(final String iamPrincipalArn) {
+        Optional<AwsIamRoleRecord> iamRole = awsIamRoleDao.getIamRole(iamPrincipalArn);
+
+        try {
```

Comment:
@tlisonbee The try/catch is a nice to have that allows any principal to authenticate, so long as it has assumed an IAM role with permissions.

I suspect that this code will more likely error due to user input, rather than a bug in this code (as it is unit tested). However, if the try portion fails because of a bug we are still logging the error and will know exactly why the code failed.